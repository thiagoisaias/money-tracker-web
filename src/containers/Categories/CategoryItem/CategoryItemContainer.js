import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { deleteCategory } from "../../../store/actions/categories/categories";

import CategoryItem from "../../../components/Categories/CategoryItem/CategoryItem";

export class CategoryItemContainer extends Component {
  handleDelete = () => {
    const { categoryData, isLoading, onDeleteCategory } = this.props;
    if (isLoading) {
      return;
    }

    if (!window.confirm("Are you sure you want to delete this category?")) {
      return;
    }
    onDeleteCategory(categoryData.id);
  };

  render() {
    const { categoryData, handleActiveItem, isActive } = this.props;
    return (
      <CategoryItem
        categoryData={categoryData}
        handleDelete={this.handleDelete}
        handleActiveItem={handleActiveItem}
        isActive={isActive}
      />
    );
  }
}

CategoryItemContainer.propTypes = {
  categoryData: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired
  }).isRequired,
  handleActiveItem: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired
};

const mapStateToProps = state => {
  return {
    isLoading: state.categories.isLoading
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onDeleteCategory: categoryId => {
      dispatch(deleteCategory(categoryId));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(
  CategoryItemContainer
);
