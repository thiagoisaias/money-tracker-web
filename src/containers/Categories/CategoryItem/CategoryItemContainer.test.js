import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { CategoryItemContainer } from "./CategoryItemContainer";

describe("CategoryItemContainer", () => {
  const mockProps = {
    categoryData: {
      id: 1,
      name: "Lorem",
      color: "#A3H7S3"
    },
    handleActiveItem: jest.fn(),
    history: {},
    isActive: false,
    isLoading: false,
    onDeleteAccount: jest.fn(),
    onSetAccountToEdit: jest.fn()
  };
  const component = shallow(<CategoryItemContainer {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
