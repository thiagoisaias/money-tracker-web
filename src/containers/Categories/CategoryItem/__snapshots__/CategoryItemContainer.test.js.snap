// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItemContainer renders properly 1`] = `
<p>
  otherValue
</p>
`;
